runcommandonset/tests/runcommandonset.set.tests.ps1

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'tests for runcommandonset set' {
+    BeforeAll {
+        $oldPath = $env:DSC_RESOURCE_PATH
+        $env:DSC_RESOURCE_PATH = Join-Path $PSScriptRoot ".."
+    }
+
+    AfterEach {
+        if (Test-Path $TestDrive/output.txt) {
+            Remove-Item -Path $TestDrive/output.txt
+        }
+    }
+
+    AfterAll {
+        $env:DSC_RESOURCE_PATH = $oldPath
+    }
+
+    It 'Input for executable and arguments can be sent to the resource' {
+        $input_json = @"
+        {
+            "executable": "pwsh",
+            "arguments": ["-Command", "echo hello world"]
+        }
+"@
+        $input_json | dsc resource set -r Microsoft.DSC.Transitional/RunCommandOnSet
+        # TODO: test output once DSC PR to capture it is merged
+        $LASTEXITCODE | Should -Be 0
+    }
+
+    It 'STDOUT captured via STDERR when calling resource directly' {
+        $input_json = @"
+        {
+            "executable": "pwsh",
+            "arguments": ["-Command", "echo hello world"]
+        }
+"@
+        $input_json | runcommandonset set 2> $TestDrive/output.txt
+        $actual = Get-Content -Path $TestDrive/output.txt
+        $actual | Should -Contain 'Stdout: hello'
+        $actual | Should -Contain 'world'
+        $LASTEXITCODE | Should -Be 0
+    }
+
+    It 'STDERR captured when calling resource directly with invalid args' {
+        $json = runcommandonset set -e pwsh -a "echo hello world" 2> $TestDrive/output.txt
+        $stdout = $json | ConvertFrom-Json
+        $stdout.exit_code | Should -Be 64
+        $expected = "Stderr: The argument 'echo hello world' is not recognized as the name of a script file. Check the spelling of the name, or if a path was included, verify that the path is correct and try again."
+        $stderr = Get-Content -Path $TestDrive/output.txt
+        $stderr | Should -Contain $expected
+        $LASTEXITCODE | Should -Be 0
+    }
+
+    It 'Executable is a required input via CLI arguments' {
+        $null = runcommandonset set -a foo
+        $LASTEXITCODE | Should -Be 4
+    }
+
+    It 'Executable is a required input via STDIN' {
+        $null = '{ "arguments": "foo" }' | dsc resource set -r Microsoft.DSC.Transitional/RunCommandOnSet
+        $LASTEXITCODE | Should -Be 2
+    }
+
+    It 'Executable can be provided without arguments' {
+        $null = '{ "executable": "pwsh" }' | dsc resource set -r Microsoft.DSC.Transitional/RunCommandOnSet
```

Comment:
instead of assigning to `$null`, you should store it in a variable and confirm that the executed binary has the expected exit code